docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,567 @@
+<li><a href="#alibaba-tab-2" data-toggle="tab">Alibaba</a></li>
+</ul>
+<br>
+<div class="tab-content">
+<div class="tab-pane fade in active" id="aws-tab-2" markdown="1">
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: aws
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+<div class="tab-pane fade" id="gcp-tab-2" markdown="1">
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: gcp
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+<div class="tab-pane fade" id="azure-tab-2" markdown="1">
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: azure
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+<div class="tab-pane fade" id="alibaba-tab-2" markdown="1">
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: alibaba
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+</div>
+
+After creating the `MySQLInstanceRequirement` Crossplane will provision a
+database instance on your provider of choice. Once provisioning is complete, you
+should see `READY: True` in the output when you run:
+```
+kubectl get mysqlinstancerequirement.database.example.org my-db
+```
+
+You should also see a `Secret` in the `default` namespace named `db-conn` that
+contains fields for `username`, `password`, and `endpoint`:
+```
+kubectl get secrets db-conn
+```
+
+## Consume Infrastructure
+
+Because connection secrets are written as a Kubernetes `Secret` they can easily
+be consumed by Kubernetes primitives. The most basic building block in
+Kubernetes is the `Pod`. Let's define a `Pod` that will show that we are able to
+connect to our newly provisioned database.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: see-db
+  namespace: default
+spec:
+  containers:
+  - name: see-db
+    image: postgres:9.6
+    command: ['psql']
+    args: ['-c', 'SELECT current_database();']
+    env:
+    - name: PGDATABASE
+      value: postgres
+    - name: PGHOST
+      valueFrom:
+        secretKeyRef:
+          name: db-conn
+          key: endpoint
+    - name: PGUSER
+      valueFrom:
+        secretKeyRef:
+          name: db-conn
+          key: username
+    - name: PGPASSWORD
+      valueFrom:
+        secretKeyRef:
+          name: db-conn
+          key: password
```

Comment:
```suggestion
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: see-db
  namespace: default
spec:
  containers:
  - name: see-db
    image: mysql:5.7
    command: ['mysql']
    args: ['-u', '$(MYSQL_USER)', '-e', 'SELECT CURRENT_USER();']
    env:
    - name: MYSQL_HOST
      valueFrom:
        secretKeyRef:
          name: db-conn
          key: endpoint
    - name: MYSQL_USER
      valueFrom:
        secretKeyRef:
          name: db-conn
          key: username
    - name: MYSQL_PWD
      valueFrom:
        secretKeyRef:
          name: db-conn
          key: password
```

This should do the trick:

```
$ kubectl logs -f see-db
CURRENT_USER()
myadmin@%
```

Reply:
Thanks @negz. Apologies for getting this mixed up the first time, and thanks for providing another solution :)